aospy/data_loader.py

```diff
@@ -31,9 +38,13 @@ def rename_grid_attrs(data):
         conventions
     """
     for name_int, names_ext in internal_names.GRID_ATTRS.items():
-        data_coord_name = set(names_ext).intersection(set(data.variables))
+        dims_and_vars = set(data.variables).union(set(data.dims))
+        data_coord_name = set(names_ext).intersection(dims_and_vars)
         if data_coord_name:
             data = data.rename({data_coord_name.pop(): name_int})
+            # Force all dimensions to have coordinates.
+            if not data[name_int].coords:
+                data = data.assign_coords(**{name_int: data[name_int].values})
```

Comment:
You can implement option (2) in an even simpler manner (rather than using ```.values``` use the full object):
```python
if not data[name_int].coords:
    data = data.assign_coords(**{name_int: data[name_int]})
```